src/pivot-table/hooks/use-column-width.ts

```diff
@@ -119,10 +124,13 @@ export default function useColumnWidth(
     });
 
     const leftGridMaxWidth = rect.width * LEFT_GRID_MAX_WIDTH_RATIO;
-    if (sumOfWidths < leftGridMaxWidth) return widths;
+    if (sumOfWidths < leftGridMaxWidth) return { widths, originalWidths: widths };
 
     const multiplier = leftGridMaxWidth / sumOfWidths;
-    return widths.map((w) => w * multiplier);
+    return {
```

Comment:
I think you can refactor this so that it returns both the `fullWidths` array and the value for `leftGridWidth`. That would mean that you don't need the `leftGridWidth` on line 179 and only need to keep one array of widths.

Reply:
nice point, actually I can calculate it once, independent of return or if statement (above them) and then only return the result in each case. (of course multiplied by .75 coefficient)

+ 

i can merge `leftGridWidth` (as you mentioned) and `getLeftGridColumnWidth` also with the return object and access those from there. a minor refactor per se!

Reply:
Maybe a dumb question but worth asking:
why do we need to loop over `widths` array to multiply each value by ratio we have? 
```
widths.reduce((totalWidth, w) => totalWidth + w * multiplier, 0)
```

why not just not multiply `sumOfWidths` we already have with ratio? or in fact we already have it there -> `leftGridMaxWidth` is just sitting there in front of us.

Reply:
Nice catch!

Reply:
Are you going to make another change here? You are currently returning two arrays of column widths

Reply:
Yes, I was thinking in addition of `actualColumnWidths` and `modifiedColumnWidths`, we can include `leftGridWidth` and `getLeftGridColumnWidth(idx)` into that useMemo since all 4 is related to the left grid, I'll update this PR soon

Reply:
I like the approach to calculate leftGridWidth in here, but you still don't need modifiedColumnWidths. if you do the `Math.min` I suggested below, you can skip all the logic from line 135 as well.